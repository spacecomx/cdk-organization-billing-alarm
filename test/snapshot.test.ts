import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import {
  MasterAccountAlarmTest,
  MasterAccountAlarmWithAwsServiceTest,
  LinkedAccountAlarmTest,
} from '../src/integ.default';

test('integration snapshot test for master account billing alarm configuation', ()=> {
  const integration = new MasterAccountAlarmTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for master account billing alarm configuation with aws service', ()=> {
  const integration = new MasterAccountAlarmWithAwsServiceTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});

test('integration snapshot test for linked account billing alarm configuation', ()=> {
  const integration = new LinkedAccountAlarmTest();

  integration.stack.forEach(stack => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});