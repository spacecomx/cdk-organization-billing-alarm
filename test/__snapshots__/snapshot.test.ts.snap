// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration snapshot test for linked account billing alarm configuation 1`] = `
Object {
  "Outputs": Object {
    "LinkedAccountAlarmBillingAlarm123456789000TopicArnDBF6D287": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "{{resolve:secretsmanager:arn:",
            Object {
              "Ref": "AWS::Partition",
            },
            ":secretsmanager:",
            Object {
              "Ref": "AWS::Region",
            },
            ":",
            Object {
              "Ref": "AWS::AccountId",
            },
            ":secret:test/billing/topicArn:SecretString:::}}",
          ],
        ],
      },
    },
    "LinkedAccountAlarmBillingAlarm444455556666TopicArnB3D60750": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "{{resolve:secretsmanager:arn:",
            Object {
              "Ref": "AWS::Partition",
            },
            ":secretsmanager:",
            Object {
              "Ref": "AWS::Region",
            },
            ":",
            Object {
              "Ref": "AWS::AccountId",
            },
            ":secret:test/billing/topicArn:SecretString:::}}",
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "LinkedAccountAlarmBillingAlarm123456789000Alarm1D6686DA": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "{{resolve:secretsmanager:arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":secretsmanager:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":secret:test/billing/topicArn:SecretString:::}}",
              ],
            ],
          },
        ],
        "AlarmDescription": "Billing Alarm for AWS DynamoDB charge estimates only (Account: 123456789000)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "123456789000",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonDynamoDB",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 120,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "LinkedAccountAlarmBillingAlarm444455556666AlarmC8ACEB09": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "{{resolve:secretsmanager:arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":secretsmanager:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ":secret:test/billing/topicArn:SecretString:::}}",
              ],
            ],
          },
        ],
        "AlarmDescription": "Consolidated billing alarm for all AWS service charge estimates (Account: 444455556666)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "LinkedAccount",
            "Value": "444455556666",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 50,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "LinkedAccountAlarmBillingAlarm444455556666TopicjohnexampleorgE304F0C9": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{{resolve:secretsmanager:arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":secretsmanager:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":secret:test/billing/topicArn:SecretString:::}}",
            ],
          ],
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for master account billing alarm configuation 1`] = `
Object {
  "Outputs": Object {
    "MasterAccountAlarmBillingAlarmTopicArn8A75BB7D": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "MasterAccountAlarmBillingAlarmTopicAFD42C2F",
      },
    },
  },
  "Resources": Object {
    "MasterAccountAlarmBillingAlarm7A68450F": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "MasterAccountAlarmBillingAlarmTopicAFD42C2F",
          },
        ],
        "AlarmDescription": "Consolidated billing alarm for all AWS Service charges",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonDynamoDB",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 140,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "MasterAccountAlarmBillingAlarmTopicAFD42C2F": Object {
      "Properties": Object {
        "DisplayName": "Organizational billing alarm topic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "MasterAccountAlarmBillingAlarmTopicjohnexampleorg46160FA3": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "MasterAccountAlarmBillingAlarmTopicAFD42C2F",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;

exports[`integration snapshot test for master account billing alarm configuation with aws service 1`] = `
Object {
  "Outputs": Object {
    "MasterAccountAlarmBillingAlarmTopicArn8A75BB7D": Object {
      "Description": "SNS Topic Arn.",
      "Value": Object {
        "Ref": "MasterAccountAlarmBillingAlarmTopicAFD42C2F",
      },
    },
  },
  "Resources": Object {
    "MasterAccountAlarmBillingAlarm7A68450F": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "MasterAccountAlarmBillingAlarmTopicAFD42C2F",
          },
        ],
        "AlarmDescription": "Billing Alarm for AWS DynamoDB charge estimates only (Account: 12345444000)",
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "Currency",
            "Value": "USD",
          },
          Object {
            "Name": "ServiceName",
            "Value": "AmazonDynamoDB",
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "EstimatedCharges",
        "Namespace": "AWS/Billing",
        "Period": 21600,
        "Statistic": "Maximum",
        "Threshold": 140,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "MasterAccountAlarmBillingAlarmTopicAFD42C2F": Object {
      "Properties": Object {
        "DisplayName": "Organizational billing alarm topic",
      },
      "Type": "AWS::SNS::Topic",
    },
    "MasterAccountAlarmBillingAlarmTopicjohnexampleorg46160FA3": Object {
      "Properties": Object {
        "Endpoint": "john@example.org",
        "Protocol": "email",
        "TopicArn": Object {
          "Ref": "MasterAccountAlarmBillingAlarmTopicAFD42C2F",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
  },
}
`;
